SWLOR.Game.Server/Feature/ShipModuleDefinition/CapitalWeaponsModuleDefinition.cs

```diff
@@ -0,0 +1,270 @@
+                {
+                    var attackRoll = Random.D10(1);
+                    var attackType = "";
+                    var attackBonus = activatorShipStatus.ThermalDamage;
+                    var attackerStat = GetAbilityScore(activator, AbilityType.Perception);
+                    var attack = Stat.GetAttack(activator, AbilityType.Perception, SkillType.Piloting, attackBonus);
+                    var defenseBonus = targetShipStatus.ThermalDefense * 2;
+                    var defense = Stat.GetDefense(target, CombatDamageType.Thermal, AbilityType.Vitality, defenseBonus);
+                    if (attackRoll < 7)
+                    {
+                        attackType = "laser";
+                    }
+                    else if (attackRoll < 9)
+                    {
+                        attackType = "ion";
+                        dmg += dmg / 3;
+                        attackBonus = activatorShipStatus.EMDamage;
+                        defenseBonus = targetShipStatus.EMDefense * 2;
+                        defense = Stat.GetDefense(target, CombatDamageType.EM, AbilityType.Vitality, defenseBonus);
+                    }
+                    else
+                    {
+                        attackBonus = activatorShipStatus.ExplosiveDamage;
+                        dmg += dmg / 4;
+                        defenseBonus = targetShipStatus.EMDefense * 2;
+                        defense = Stat.GetDefense(target, CombatDamageType.Explosive, AbilityType.Vitality, defenseBonus);
+                        if (targetShipStatus.Shield <= 4)
+                        {
+                            dmg += dmg / 4;
+                        }
+                    }
+                    var defenderStat = GetAbilityScore(target, AbilityType.Vitality);
+                    var damage = Combat.CalculateDamage(
+                        attack,
+                        dmg,
+                        attackerStat,
+                        defense,
+                        defenderStat,
+                        0);
+
+                    var chanceToHit = Space.CalculateChanceToHit(activator, target);
+                    var roll = Random.D100(1);
+                    var isHit = roll <= chanceToHit;
+
+                    var sound = EffectVisualEffect(VisualEffect.Vfx_Ship_Blast);
+                    var missile = EffectVisualEffect(VisualEffect.Mirv_StarWars_Bolt2);
+
+                    if (attackType == "laser")
+                    {
+                        if (isHit)
+                        {
+                            AssignCommand(activator, () =>
+                            {
+                                ApplyEffectToObject(DurationType.Instant, sound, target);
+                                ApplyEffectToObject(DurationType.Instant, missile, target);
+
+                                DelayCommand(0.3f, () =>
```

Comment:
What's the purpose of the delay?

Reply:
The delay is the same as for combat lasers I believe (on phone atm hard to double check). Just has the damage and other effects occur after allowing time for the vfx to conclude.